service/k8sclient/k8sclient_test.go

```diff
@@ -39,6 +41,140 @@ import (
 	"github.com/percona-platform/dbaas-controller/utils/logger"
 )
 
+// VersionServiceClient represents a client for Version Service API.
+type VersionServiceClient struct {
+	url  string
+	http *http.Client
+}
+
+// NewVersionServiceClient creates a new client for given version service URL.
+func NewVersionServiceClient(url string) *VersionServiceClient {
+	return &VersionServiceClient{
+		url: url,
+		http: &http.Client{
+			Timeout: 10 * time.Second,
+		},
+	}
+}
+
+// componentsParams contains params to filter components in version service API.
+type componentsParams struct {
+	product        string
+	productVersion string
+}
+
+type matrix struct {
+	PXCOperator   map[string]interface{} `json:"pxcOperator,omitempty"`
+	PSMDBOperator map[string]interface{} `json:"psmdbOperator,omitempty"`
+}
+
+type Version struct {
+	Product        string `json:"product"`
+	ProductVersion string `json:"operator"`
```

Comment:
Is that correct? The field is `ProductVersion`, but annotation `operator`

Reply:
Yes, it is correct. The version service was designed only with operators in mind. Then we decided to include PMM <-> operator dependencies. I did not change it to be backwards compatible.